src/cast.rs

```diff
@@ -589,3 +647,17 @@ fn as_primitive() {
     let x: u8 = (768i16).as_();
     assert_eq!(x, 0);
 }
+
+#[test]
+fn float_to_integer_checks_overflow() {
+    // This will overflow an i32
+    let source: f64 = 1.0e+123f64;
+
+    // Expect the overflow to be caught
+    assert_eq!(source.to_i32(), None);
+
+    // Specifically make sure we didn't silently let the overflow through
+    // (This line is mostly for humans -- the robots should catch any problem
+    // on the line above).
+    assert!(source.to_i32() != Some(-2147483648));
```

Comment:
As a human, I don't understand the point of this line.  If it was equal to `None` above, then it won't be equal to _any_ `Some`.  Trust the "robots".

Also note that -2147483648 is exactly `i32::MIN`, which would be a perfectly legal value!  It just happens that LLVM is giving you this value for overflow too, but that's UB.

Reply:
I figured it might be nice to have some visual correspondence between this test and the PR, but I agree that it's "nonsensical" otherwise -- we shouldn't ever get to this line. I've removed it.